hail/src/main/scala/is/hail/types/physical/stypes/SCode.scala

```diff
@@ -112,6 +112,8 @@ trait SValue {
   def st: SType
 
   def get: SCode
+
+  def hash(cb: EmitCodeBuilder): SInt32Code = throw new IllegalArgumentException
```

Comment:
This should be an `UnsupportedOperationException`, and it should have a message when you construct it.